Components/Metrics/PointToSurfaceDistance/itkPointToSurfaceDistanceMetric.h

```diff
@@ -0,0 +1,126 @@
+    using PointDataIterator = typename Superclass::PointDataIterator;
+    using InputPointType = typename Superclass::InputPointType;
+    using OutputPointType = typename Superclass::OutputPointType;
+    using CoordRepType = typename OutputPointType::CoordRepType;
+    using VnlVectorType = vnl_vector< CoordRepType >;
+    using NonZeroJacobianIndicesType = typename Superclass::NonZeroJacobianIndicesType;
+
+	   /** Constants for the pointset dimensions. */
+    itkStaticConstMacro( FixedPointSetDimension, unsigned int, TFixedPointSet::PointDimension );
+
+    using ImageType = itk::Image< float, itkGetStaticConstMacro( FixedPointSetDimension )>;
+    using InputImageType = itk::Image< unsigned char,  itkGetStaticConstMacro( FixedPointSetDimension ) >;
+    using RescalerType = itk::RescaleIntensityImageFilter< ImageType, ImageType >;
+    using DtFilterType = itk::DanielssonDistanceMapImageFilter< InputImageType, ImageType, ImageType >;
+    using NegateFilterType = itk::InvertIntensityImageFilter< InputImageType, InputImageType >;
+    using AddImageFilt = itk::AddImageFilter< ImageType, ImageType, ImageType >;
+    using SegReaderType =itk::ImageFileReader< InputImageType >;
+    using DTReaderType = itk::ImageFileReader< ImageType  >;
+    using WriterType = itk::ImageFileWriter< ImageType >;
+    using InterpolatorType = itk::BSplineInterpolateImageFunction< ImageType, float, float >;
+    using ConstDTimage = typename ImageType::ConstPointer;
+	 /** Method for creation through the object factory. */
+    itkNewMacro( Self );//OK
+
+    /** Run-time type information (and related methods). */
+    itkTypeMacro( PointToSurfaceDistanceMetric, SingleValuedPointSetToPointSetMetric );//OK
+    /**  Initialize. */
+    virtual void Initialize() noexcept(false);
+
+    /**  Get the value for single valued optimizers. */
+    MeasureType GetValue( const TransformParametersType & parameters ) const noexcept(false);
+
+    /** Get the derivatives of the match measure. */
+    void GetDerivative( const TransformParametersType & parameters, DerivativeType & Derivative ) const noexcept(false);
+
+    /**  Get value and derivatives for multiple valued optimizers. */
+    void GetValueAndDerivative( const TransformParametersType & parameters, MeasureType & Value, DerivativeType & Derivative ) const noexcept(false);
+
+    /** Set input Segmented Image File **/
+    void SetSegImageIn( const std::string  str ) ;
+
+    /** Set input Segmented Image File **/
+    void SetDTImageIn( const std::string  str ) ;
+
+    /** Set output Distance Transform File **/
+    void SetDTImageOut(  const std::string  str );
+
+    /** Get Distance Transform Image*/
+    ConstDTimage GetDTImage() const;
+
+protected:
+
+    PointToSurfaceDistanceMetric();
+    virtual ~PointToSurfaceDistanceMetric() {}
+    bool m_AvPointWeigh;
+
+private:
+
+    PointToSurfaceDistanceMetric( const Self & );
+    std::string m_SegFileIn,m_DTFileIn,m_DTFileOut;
```

Comment:
Please avoid abbreviations in C++ identifiers, if you can. "Seg", "DT", "AvPointWeigh" may be clear enough to you, but I would have to guess, or figure out the meaning.

Reply:
I have changed the names. So more manifest names have been coined.

Reply:
Thank you @gokhangg